src/app/wallets/add-invoice-for-wallet.ts

```diff
@@ -24,33 +29,32 @@ export const addInvoiceForSelf = async ({
   const wallets = WalletsRepository()
   const wallet = await wallets.findById(walletIdChecked)
   if (wallet instanceof Error) return wallet
-
   const limitOk = await checkSelfWalletIdRateLimits(wallet.accountId)
   if (limitOk instanceof Error) return limitOk
 
-  let addInvoiceWithAmount: (
-    args: AddInvoiceArgs,
-  ) => Promise<LnInvoice | ApplicationError>
-
-  switch (wallet.currency) {
-    case WalletCurrency.Btc:
-      addInvoiceWithAmount = addInvoiceSatsDenomiation
-      break
-    case WalletCurrency.Usd:
-      addInvoiceWithAmount = addInvoiceFiatDenomiation
-      break
-  }
-
-  const walletInvoiceFactory = WalletInvoiceFactory({
-    walletId: walletIdChecked,
-    currency: wallet.currency,
-  })
+  const walletInvoiceBuilder = createWalletInvoiceBuilder()
 
-  return addInvoiceWithAmount({
-    walletInvoiceCreateFn: walletInvoiceFactory.createForSelf,
-    amount,
-    memo,
-  })
+  if (walletInvoiceBuilder instanceof Error) return walletInvoiceBuilder
+
+  const walletInvoiceBuilderWithAmount = await walletInvoiceBuilder
+    .withDescription({description: memo})
```

Comment:
_assuming_ there is a single argument that can be passed to withDescription, we could just pass `(memo)`. this is a pattern we have been using where a single argument doesn't take an object but a scalar.

Reply:
nevermind